### DISCUSSION TITLE
State types (INPROGRESS, FINISHED, etc.) can't really be used with user cards

### DISCUSSION THREAD

**Discussion by AlexGuironnetRTE** [opfab/operatorfabric-core#1451] [Q&A] [answered]
Hello,

I'm working on #1430, which aims to display the state type (IN PROGRESS, CANCELED or FINISHED) instead of the state name in the card detail header.
In order to test it, I thought I'd created a new usercard bundle that would fit the following use case:
- User sends a card saying "Process XXX is started, please tell us if you see any side effect" with a text-box below (with state type INPROGRESS)
- Once the process has completed successfully, the user updates the card to say "Process XXX has completed successfully, you shoudn't see any interference anymore" with state type "FINISHED"

Note: For my purpose, the whole thing could be informative only, but currently the header is only set to appear for question cards (ie states for which the response property is set).

But unless I'm mistaken, this is not doable using user cards: as the state type is defined in the bundle for a given state, and the state is selected when creating the user card (and can't be changed afterwards), the state type is effectively fixed for a given usercard. 
And creating a new usercard for the update doesn't work either because it wouldn't replace the first one.

Am I missing something? What do you think?

**Comment by freddidierRTE** [comment, marked as answer]
Indeed when the user edit a card he can change the state (but not the process)

**Reply by AlexGuironnetRTE** [reply]
My mistake... I tested with the task reminder user card, for which the process has only one state so the state dropdown isn't shown, and assumed it meant it couldn't be changed. Thanks!

**Reply by AlexGuironnetRTE** [reply]
Come to think of it, shouldn't we rework our usercard example that's currently under Service "Base Example" and State "Process Example"?  It lets you choose a "State of the Process" and "Status of the state" that are just template values and are different from the State type property? It might be a bit  confusing.

We could demo the same use case but make it a separate process, and each "State of the Process" ("Started", "Calculation 1 done", "Calculation 2 done" etc.) would be a state in the OpFab sense, with a given type ("IN PROGRESS", "CANCELED" etc.)
I think this would fit better with the way processes are modeled in OpFab and it would make this example more interesting in the monitoring screen for example.
